components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/internal/EventUtil.java

```diff
@@ -0,0 +1,220 @@
+
+        Map<String, Object> properties = baseTenantProperties(tenantDomain);
+        properties.put(CONSUMER_KEYS, Collections.singletonList(consumerKey));
+        properties.put(IdentityEventConstants.EventProperty.APPLICATION_ID, applciationResourceId);
+        properties.put(IdentityEventConstants.EventProperty.APPLICATION_NAME, appName);
+        properties.put(EVENT_CAUSE, "Revoking the tokens of the application " + appName);
+        publish(properties);
+    }
+
+    /**
+     * Publishes a token revoke event for a specific tenant domain and consumer key.
+     *
+     * @param tenantDomain    Tenant domain of the application.
+     * @param consumerKey     Consumer key of the application.
+     * @param serviceProvider ServiceProvider object containing application details.
+     */
+    public static void publishTokenRevokeEvent(String tenantDomain, String consumerKey,
+                                               ServiceProvider serviceProvider) {
+
+        Map<String, Object> properties = baseTenantProperties(tenantDomain);
+        if (serviceProvider != null) {
+            properties.put(CONSUMER_KEYS, Collections.singletonList(consumerKey));
+            properties.put(IdentityEventConstants.EventProperty.APPLICATION_ID,
+                    serviceProvider.getApplicationResourceId());
+            properties.put(IdentityEventConstants.EventProperty.APPLICATION_NAME, serviceProvider.getApplicationName());
+            properties.put(EVENT_CAUSE,
+                    "Revoking the tokens of the application " + serviceProvider.getApplicationName());
+        }
+        publish(properties);
+    }
+
+    /**
+     * Publishes a token revoke event for a specific tenant ID and user store name.
+     *
+     * @param tenantId      Tenant ID of the user store.
+     * @param userStoreName Name of the user store.
+     * @param tokens        Set of AccessTokenDO objects representing the revoked tokens.
+     */
+    public static void publishTokenRevokeEvent(int tenantId, String userStoreName, Set<AccessTokenDO> tokens) {
+
+        Map<String, Object> properties = new HashMap<>();
+        setFirstTokenUserProperties(properties, tokens);
+        properties.put(IdentityEventConstants.EventProperty.USER_STORE_DOMAIN, userStoreName);
+        properties.put(IdentityEventConstants.EventProperty.TENANT_ID, tenantId);
+        properties.put(EVENT_CAUSE, "Revoking the tokens in the user store " + userStoreName);
+        publish(properties);
+    }
+
+    /**
+     * Publishes a token revoke event for a specific tenant ID and set of AccessTokenDO objects.
+     *
+     * @param tenantId Tenant ID of the user store.
+     * @param tokens   Set of AccessTokenDO objects representing the revoked tokens.
+     */
+    public static void publishTokenRevokeEvent(int tenantId, Set<AccessTokenDO> tokens) {
+
+        Map<String, Object> properties = new HashMap<>();
+        setFirstTokenUserProperties(properties, tokens);
```

Comment:
Well rather than taking these from AccessTokenDO don't we have a better way to get these like token message context. If it's not reliable and changing a lot in the flow and AccessTokenDO is the model used to build the token eventually it makes sense